### DISCUSSION TITLE
[RF5C64] SampleRAM switching

### DISCUSSION THREAD

**Discussion by freq-mod** [LTVA1/furnace#121] [Ideas]
Suggested to Furnace-A, got forgotten since literally one cares about the Mega-CD sound source.

RF5C64 on MCD anf FMT has just 64 kb of sampleRAM, which is almost nothing, What most MegaCD games that used RF5 did, was to load 8-bit PCCM samples into MCD main RAM (256 kB), then dynamically switch/load them during playback. It's not a demoscene technique, VGM format support it without sweating. Just a cool thing we can consider after FT import is done.
![image](https://github.com/LTVA1/furnace/assets/32672779/dda587ec-af3d-4806-88c4-3f0dc8a34ff3)
![image](https://github.com/LTVA1/furnace/assets/32672779/392b80f3-36a5-4c29-ac7a-44eed1b2157b)

**Comment by LTVA1** [comment]
So it's possible on all the devices where RF5C64 is used (since I rn know only one)? I may add this yeah, but I need to know if I would need to make 64<->256 KiB switch or it can be just given 256 KiB permanently. The problem would be that if you have e.g. several long samples that don't fit into RF5C64 RAM at once, it won't play properly on real hardware, and idk how to deal with it

Also damn why I don't receive notifications about new discussions

**Reply by freq-mod** [reply]
I would have it as a flag, and enforce max sample length of 64 kb. And RF5 was used in Sega Mega CD, Fujitsu FM Towns and Sega System 18/32 if im not mistaken (OH wait different FM towns models have different RAM sizes, og model had 2 MB)

**Reply by LTVA1** [reply]
I did not mean max sample length of 64 KiB, I meant the situation when you have two samples, 50 KiB each, playing at the same time. It would break the playback so somehow such conflicts should be resolved

**Reply by freq-mod** [reply]
Looking at ValleyBell's message (copying into RF5 RAM **in blocks of 512b**, i think it isnt much of a problem. Hearing Sonic CD past tracks(https://www.youtube.com/playlist?list=PLRCH7jfAKlyBc-OiXyaLHsb5Y0x19xBJc), it may work like that, with alternative apporoach just enforce max 64 kb at once on playback, with 256 kb max for song total.

**Reply by LTVA1** [reply]
But how you gonna enforce 64 KiB at once limit? You would need to basically write similar driver in Furnace from scratch. Idk if I would manage to do it
